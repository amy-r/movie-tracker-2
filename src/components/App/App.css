.App {
  display: flex;
  flex-direction: column;
  text-align: center;
  font-family: 'Roboto', sans-serif;
  /*display: grid;
  grid-template-columns: repeat(12, 1fr);*/
  background-color: black;
}

.header {
  background-color: red;
  height: 50px;
  width: 100%;
}

.header h1 {
  text-decoration: off;
  font-size: 1.5em;
  display: inline-block;
  margin: 10px;
  font-family: 'Roboto', sans-serif;
  font-weight: 800;
}

/*h1 .movie {
  color: brown;
  font-size: 1.7em;
}*/

.nav {
  float: right;
  padding: 5px;
  margin: 10px;
  text-decoration: off;
}
