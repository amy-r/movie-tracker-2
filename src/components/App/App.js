import React, { Component } from 'react';
import { addMovies } from '../../actions/actions';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom'

import Header from '../Header/Header';
import MovieIndex from '../Movies/MovieIndex';
import User from '../User/User';

import api from '../../apiCalls';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount = async () => {
    this.fetchMovies();
  };

  fetchMovies = async () => {
    const testRun = await api.fetchParse(api.test);
    const moviesArray = await api.movieCleaner(testRun);
    console.log(moviesArray)
    await this.props.addMovies(moviesArray);
  };

  render() {
   
    return (
      <div>
        <Header />
        <Route exact path='/' component={MovieIndex} />
        <Route exact path='/user' component={User} /> 
      </div>
    );
  }
}

const mapStateToProps = store => ({
  movies: store.movieData
});

const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch(addMovies(movies))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
