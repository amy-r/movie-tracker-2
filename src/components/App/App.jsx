import React, { Component } from 'react';
import { Route, NavLink, withRouter } from 'react-router-dom';
import { addMovies } from '../../actions/actions';
import MovieIndex from '../Movies/MovieIndex';
import * as api from '../../apiCalls';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import User from '../User/User';
import NewUser from '../NewUser/NewUser';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {};
  }


  componentDidMount = async () => {
    this.fetchMovies();
  };

  componentWillReceiveProps = () => {
    
  }
  
  fetchMovies = async () => {
    const testRun = await api.fetchParse(api.test);
    const moviesArray = await api.movieCleaner(testRun);
    await this.props.addMovies(moviesArray);
  };


  render() {
    return (
      <div className="wrapper">
        <header className="header">
          <h1>
            <span className="movie">Movie</span> Tracker
          </h1>
          <NavLink to="/login" className="nav">
            Sign in
          </NavLink>
          <NavLink to="/new-user" className="nav">
            Create an Account
          </NavLink>
        </header>
        <Route exact path="/" component={MovieIndex} />
        <Route path="/login" component={User} />
        <Route path="/new-user" component={NewUser} />
      </div>
    );
  }
}

export const mapStateToProps = store => ({
  movies: store.movieData,
  user: store.user
});

export const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch(addMovies(movies))
});

App.propTypes = {
  addMovies: PropTypes.func
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
