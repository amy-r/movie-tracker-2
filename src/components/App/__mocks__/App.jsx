
/* eslint-disable */

import React, { Component } from 'react';
import { addMovies } from '../../actions/actions';
import { connect } from 'react-redux';
import { Route, NavLink, Switch, withRouter } from 'react-router-dom';
import MovieIndex from '../Movies/MovieIndex';
import User from '../User/User';
import { NewUser } from '../NewUser/NewUser';
import * as api from '../../apiCalls';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidMount = async () => {
    this.fetchMovies();
  };

  fetchMovies = async () => {
    api.fetchParse(url);
    api.movieCleaner(object);
    return {
      title: 'title'
      poster: 'poster'
    }
  };

  render() {
    return (
      <div className="wrapper">
        <header className="header">
          <h1>
            <span className="movie">Movie</span> Tracker
          </h1>
          <NavLink to="/login" className="nav">
            Sign in
          </NavLink>
          <NavLink to="/new-user" className="nav">
            Create an Account
          </NavLink>
        </header>
        <Route exact path="/" component={MovieIndex} />
        <Route path="/login" component={User} />
        <Route path="/new-user" component={NewUser} />
      </div>
    );
  }
}

const mapStateToProps = store => ({
  movies: store.movieData
});

const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch(addMovies(movies))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
