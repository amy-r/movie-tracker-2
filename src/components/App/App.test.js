import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { shallow } from 'enzyme';

describe('APP', () => {
  it.skip('should match the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should define movies props for the container', () => {
    const title = 'RockyIV';
    const poster = '/jeffIsGreat';
    const mockStore = { movieData: [{ title, poster }] };
    const expected = [{ poster: '/jeffIsGreat', title: 'RockyIV' }];
    const mapped = mapStateToProps(mockStore);
    expect(mapped.movies).toEqual(expected);
  });

  it('should define user props for the container', () => {
    const id = 0;
    const name = 'jeffIsGreat';
    const mockStore = { user: {id, name} };
    const expected = { id: 0, name: 'jeffIsGreat' };
    const mapped = mapStateToProps(mockStore);
    expect(mapped.user).toEqual(expected);
  });

  it('should call the dispatch function on MDTP', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addMovies();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
