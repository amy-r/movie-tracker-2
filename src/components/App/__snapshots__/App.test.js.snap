// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    favorites={Array []}
    movies={Array []}
    user={
        Object {
            "id": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="title"
                    to="/"
          >
                    <h1>
                              Movie Tracker
                    </h1>
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/login"
          >
                    Sign in
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/new-user"
          >
                    Create an Account
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    onClick={undefined}
                    to="/login"
          >
                    Sign out
          </NavLink>
</header>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          path="/login"
/>,
        <Route
          component={[Function]}
          path="/new-user"
/>,
        <Route
          component={[Function]}
          path="/favorites"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="title"
              to="/"
>
              <h1>
                            Movie Tracker
              </h1>
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/login"
>
              Sign in
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/new-user"
>
              Create an Account
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              onClick={undefined}
              to="/login"
>
              Sign out
</NavLink>,
            null,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <h1>
                Movie Tracker
</h1>,
              "className": "title",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movie Tracker",
              },
              "ref": null,
              "rendered": "Movie Tracker",
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Sign in",
              "className": "nav",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Sign in",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Create an Account",
              "className": "nav",
              "to": "/new-user",
            },
            "ref": null,
            "rendered": "Create an Account",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Sign out",
              "className": "nav",
              "onClick": undefined,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Sign out",
            "type": [Function],
          },
          null,
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/new-user",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="title"
                        to="/"
            >
                        <h1>
                                    Movie Tracker
                        </h1>
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/login"
            >
                        Sign in
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/new-user"
            >
                        Create an Account
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        onClick={undefined}
                        to="/login"
            >
                        Sign out
            </NavLink>
</header>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            path="/login"
/>,
          <Route
            component={[Function]}
            path="/new-user"
/>,
          <Route
            component={[Function]}
            path="/favorites"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="title"
                to="/"
>
                <h1>
                                Movie Tracker
                </h1>
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/login"
>
                Sign in
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/new-user"
>
                Create an Account
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                onClick={undefined}
                to="/login"
>
                Sign out
</NavLink>,
              null,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <h1>
                  Movie Tracker
</h1>,
                "className": "title",
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Movie Tracker",
                },
                "ref": null,
                "rendered": "Movie Tracker",
                "type": "h1",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Sign in",
                "className": "nav",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Sign in",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Create an Account",
                "className": "nav",
                "to": "/new-user",
              },
              "ref": null,
              "rendered": "Create an Account",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Sign out",
                "className": "nav",
                "onClick": undefined,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Sign out",
              "type": [Function],
            },
            null,
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/new-user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
